### PR TITLE
[GeoMechanicsApplication] Add abc model 1D column test cases to our test suite

### LINKED ISSUE #13713: [GeoMechanicsApplication] Add abc model 1D column test cases to our test suite
After the investigations, it would be nice to add at least two (dry and fully saturated) test cases of the investigated ones to our test suite + Documentation with nice pictures and explanations

### PR DESCRIPTION
**Pull request by mnabideltares**
**📝 Description**
Test case for fully saturated colomn with settlement

**🆕 Changelog**
- Added test case
- Added documentation

### CODE REVIEW

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/MaterialParameters_abc.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          "PERMEABILITY_XX": 1.15741E-05,
+          "PERMEABILITY_YY": 1.15741E-05,
+          "PERMEABILITY_XY": 0,
+          "DYNAMIC_VISCOSITY": 10000,
+          "RETENTION_LAW": "SaturatedBelowPhreaticLevelLaw",
+          "RESIDUAL_SATURATION": 0.06203,
+          "SATURATED_SATURATION": 1,
+          "VAN_GENUCHTEN_AIR_ENTRY_PRESSURE": 2.561,
+          "VAN_GENUCHTEN_GN": 1.377,
+          "VAN_GENUCHTEN_GL": 1.25,
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/MaterialParameters_abc.json]: Please removed these van Genuchten parameters, as they are not used.
If the purpose of this test is the always fully saturated soil, we could also use SaturatedLaw.

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/MaterialParameters_linear.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          "PERMEABILITY_XX": 1.15741E-05,
+          "PERMEABILITY_YY": 1.15741E-05,
+          "PERMEABILITY_XY": 0,
+          "DYNAMIC_VISCOSITY": 10000,
+          "RETENTION_LAW": "SaturatedBelowPhreaticLevelLaw",
+          "RESIDUAL_SATURATION": 0.06203,
+          "SATURATED_SATURATION": 1,
+          "VAN_GENUCHTEN_AIR_ENTRY_PRESSURE": 2.561,
+          "VAN_GENUCHTEN_GN": 1.377,
+          "VAN_GENUCHTEN_GL": 1.25,
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/MaterialParameters_linear.json]: Please remove the not used van Genuchten parameters

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+        "python_module"            : "fix_model_part_above_phreatic_line_process",
+        "kratos_module"            : "KratosMultiphysics.GeoMechanicsApplication",
+        "process_name"             : "FixWaterPressuresAbovePhreaticLineProcess",
+        "Parameters": {
+          "model_part_name"        : "PorousDomain.SoilExcludingBoundary",
+          "x_coordinates"          : [0, 1],
+          "y_coordinates"          : [50, 50],
+          "z_coordinates"          : [0, 0]
+        }
+      },
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json]: If the column always is fully saturated, no part of the column will be above the phreatic line and this process could be omitted. That applies for all stages of the computation.

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+    "constraints_process_list": [
+      {
+        "python_module"            : "apply_scalar_constraint_table_process",
+        "kratos_module"            : "KratosMultiphysics.GeoMechanicsApplication",
+        "process_name"             : "ApplyScalarConstraintTableProcess",
+        "Parameters": {
+          "model_part_name"        : "PorousDomain.TopWaterPressure0",
+          "variable_name"          : "WATER_PRESSURE",
+          "is_fixed"               : true,
+          "value"                  : 0.0,
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json]: For the apply_scalar_constraint_table_process with fluid_pressure_type Phreatic_Multi_Line the x_ y_ and z_coordinaters are used, this "value" is not and can be omitted.

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/README.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# Settlement on a fully saturated column with uniform load
+This test consists of a rectangular soil domain, 1.0m wide and 50.0m deep. The mesh consists of SmallStrainUPwDiffOrderElement2D6N elements. The pheratic line is kept at the top of the column, namely at $y = 50$ [m], resulting in a column with hydrostatic pressure. A schematic can be found in the figure below:
+![Schematic](Schematic.svg)
+
+## Setup
+
+The test is performed in 5 stages:
+1. A K0 stage with a linear elastic model, with a Young's modulus of 1e9 [Pa] and a Poisson ratio of 0.2.
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/README.md]: Maybe clearer for many to give Young's modulus in MPa or GPa i.s.o. Pa
Poisson ratio -> Poisson's ratio

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/README.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+2. A settlement stage with abc model (100 days or 8640000 seconds). The keywords "reset_displacement" is set to true here, such that the total displacements start counting from the start of this stage.
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/README.md:13]: We indeed use the dll for the abc model. However with the other settings of "USE_HENCKY_STRAIN": false and MOVE_MESH_FLAG false it is a NEN_Bjaerum formulation. In the very first paragraph # Settlement it is wise to mention the material models used too.

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/README.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+3. Another settlement stage (8.64 seconds), the same uniform load and also the same material as previous stage is applied. The time-step is 0.001 days (8.64 seconds), such that there is horizontal stress distribution by the abc model. 
+4. A second settlment stage (0.1 day) with a uniform line-load of 20 [kN] applied in the negative Y direction to the top of the model.
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/README.md]: Are we very sure of this staging? stage 3 and stage 4 seem both short. It is strange for me that horizontal distribution would happen suddenly after 100 days of stage 2. horizontal distribution of the new load in stage 4 seems to be omitted here. The way it is presented, stage 4 and 5 could be together in one, with a table for the load application.

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/README.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+5. A second settlement stage, such that the total time reaches 10000 days (or 864000000 seconds). The same uniform load of 20 [kN] is applied in the negative Y direction. The time-step is > 0.001 days, such that there is no horizontal stress distribution by the abc model.
+
+The following common conditions hold for all stages:
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/README.md:18]: The top and bottom Dirichlet condition for water pressure are not described here. The LineNormaLoadDiffOrderCondition2D3N for water weight above the soil surface could be omitted here ( and in the .json files ), because the  phreatic waterlevel remains exactly at the top of the soil. Hence no weight of water mass above the top of the soil.

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/Schematic.svg
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/Schematic.svg:1]: The displacement boundary conditions are nicely included in this picture. Water pressure boundary conditions are omitted. Top and bottom are permeable, sides are closed. I sadly do not know about clear symbols for that, but guess that they exist.

File: applications/GeoMechanicsApplication/tests/test_dsettlement_validation.py [hunk 9]
```diff
@@ -61,6 +61,61 @@ def test_settlement_dry_column(self):
...
+        import KratosMultiphysics.GeoMechanicsApplication.run_geo_settlement as run_geo_settlement
+
+        n_stages = 5
+        project_parameters_filenames = [
+            f"ProjectParameters_stage{i+1}.json" for i in range(n_stages)
+        ]
+        status = run_geo_settlement.run_stages(
+            project_path, project_parameters_filenames
+        )
+        self.assertEqual(status, 0)
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/test_dsettlement_validation.py:87]: The setup implies that when a later stage fails to run, results of previously completed stages are not checked.

File: applications/GeoMechanicsApplication/tests/test_dsettlement_validation.py [hunk 10]
```diff
@@ -61,6 +61,61 @@ def test_settlement_dry_column(self):
...
+
+        reader = test_helper.GiDOutputFileReader()
+
+        output_data = reader.read_output_from(
+            os.path.join(project_path, "stage2.post.res")
+        )
+        actual_settlement_after_one_hundred_days = reader.nodal_values_at_time(
+            "TOTAL_DISPLACEMENT", 8640000, output_data, [104]
+        )[0][1]
+        expected_settlement_after_one_hundred_days = -1.71094
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/test_dsettlement_validation.py]: like this the expected result is explicitely named. For me it is equally clear to enter the numerical value at line 100 in the assertAlmostEqual statement. A matter of preference I suppose.

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/mesh.mdpa [hunk 11]
```diff
@@ -0,0 +1,6532 @@
...
+
+Begin SubModelPart Bottom1
+  Begin SubModelPartNodes
+    1
+    4
+    204
+  End SubModelPartNodes
+End SubModelPart 
+
+Begin SubModelPart FixedPressure2
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/mesh.mdpa]: These names originate directly from DSettlement gui. They are not very descriptive. FixedPressure2 appears to be the nodes for fixing water pressure on the bottom of the column. I have discussed with Walter today to make these names more explanatory, such that we spend less time on understanding the setup next time.

**Review by WPK4FEM — CHANGES_REQUESTED** [review]
Hi Mohamed,
Thank you for the addition to the test suite. Almost ready to merge. lets discuss a little bit in person the readme text describing the test.

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage2.json [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+    "parallel_type"                   : "OpenMP",
+    "number_of_threads"               : 1
+  },
+  "solver_settings": {
+    "solver_type"                     : "U_Pw",
+    "model_part_name"                 : "PorousDomain",
+    "domain_size"                     : 2,
+    "model_import_settings": {
+      "input_type"                    : "mdpa",
+      "input_filename"                : "mesh"
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage2.json]: Here we can use the following to make sure we don't re-read the mesh (same holds for stage 3-5)
```suggestion
      "input_type"                    : "use_input_model_part"
```

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json [hunk 13]
```diff
@@ -0,0 +1,251 @@
...
+    "solution_type"                   : "Quasi-Static",
+    "scheme_type"                     : "Backward_Euler",
+    "reset_displacements"             : true,
+    "strategy_type"                   : "line_search",
+    "convergence_criterion"           : "displacement_criterion",
+    "displacement_relative_tolerance" : 0.001,
+    "displacement_absolute_tolerance" : 1.0E-9,
+    "min_iterations"                  : 6,
+    "max_iterations"                  : 30,
+    "number_cycles"                   : 100,
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json]: I find this quite a high number, I think even 1 should work right, since this converges quickly anyway (same holds for the other stages)?
```suggestion
    "number_cycles"                   : 1,
```

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json [hunk 14]
```diff
@@ -0,0 +1,251 @@
...
+    "reduction_factor"                : 0.5,
+    "increase_factor"                 : 2.0,
+    "desired_iterations"              : 4,
+    "max_radius_factor"               : 10.0,
+    "min_radius_factor"               : 0.1,
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json]: I don't think we need these, since we're not using arc-length (same holds for other stages)
```suggestion
```

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json [hunk 15]
```diff
@@ -0,0 +1,251 @@
...
+                                           "WATER_PRESSURE",
+                                           "VOLUME_ACCELERATION",
+                                           "LINE_LOAD_Y",
+                                           "TOTAL_STRESS_TENSOR",
+                                           "CAUCHY_STRESS_TENSOR"],
+              "gauss_point_results"     : ["GREEN_LAGRANGE_STRAIN_TENSOR",
+                                           "ENGINEERING_STRAIN_TENSOR",
+                                           "VON_MISES_STRESS",
+                                           "FLUID_FLUX_VECTOR",
+                                           "HYDRAULIC_HEAD"]
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json]: Could you bring the number of outputs down to the ones we assert in the test case?

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json [hunk 16]
```diff
@@ -0,0 +1,251 @@
...
+    "problem_domain_sub_model_part_list": ["Soil-0"],
+    "processes_sub_model_part_list"   : ["gravity",
+                                         "PhreaticLineContactStress0",
+                                         "TopWaterPressure0",
+                                         "Bottom1",
+                                         "FixedPressure2",
+                                         "Side3",
+                                         "Side4",
+                                         "K0_model_part",
+                                         "Head_line0"],
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json]: To me not all names of the model parts are completely clear and might be clarified a bit (e.g. Side3/Side4 could be LeftSide/RightSide etc)

File: applications/GeoMechanicsApplication/tests/test_dsettlement_validation.py (hunk 9 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/test_dsettlement_validation.py:87]: I think that depends on if 'run_geo_settlement' does an early return (i.e. only returns 0 when all stages are fine), or returns the status of the last stage only

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/README.md [hunk 17]
```diff
@@ -0,0 +1,30 @@
+# Settlement on a fully saturated column with uniform load
+This test consists of a rectangular soil domain, 1.0m wide and 50.0m deep. The mesh consists of SmallStrainUPwDiffOrderElement2D6N elements. The pheratic line is kept at the top of the column, namely at $y = 50$ [m], resulting in a column with hydrostatic pressure. Here, the meterial property "USE_HENCKY_STRAIN" is set to false, in combination with "MOVE_MESH_FLAG" as false, it leads to a NEN-Bjerrum formulation.
```
Comment (rfaasse) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/README.md]: Could you fix these minor typos
```suggestion
This test consists of a rectangular soil domain, 1.0m wide and 50.0m deep. The mesh consists of SmallStrainUPwDiffOrderElement2D6N elements. The phreatic line is kept at the top of the column, namely at $y = 50$ [m], resulting in a column with hydrostatic pressure. Here, the material property "USE_HENCKY_STRAIN" is set to false, in combination with "MOVE_MESH_FLAG" as false, it leads to a NEN-Bjerrum formulation.
```

**Review by rfaasse — COMMENTED** [review]
Hi Mohamed, thanks for making the effort of cleaning up the input files, they're very readable! I have a couple of minor suggestions for improvements.

I was doubting whether we might want to combine the README of this one with testcase1, but seeing this, I think it's nicer to have the test folders being self-explanatory like they are now 👍

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/MaterialParameters_abc.json [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+          "BIOT_COEFFICIENT": 1,
+          "DENSITY_SOLID": 2242.609582059123,
+          "DENSITY_WATER": 1019.367991845056,
+          "POROSITY": 0.5,
+          "BULK_MODULUS_SOLID": 20000000000,
+          "BULK_MODULUS_FLUID": 2200000000,
+          "PERMEABILITY_XX": 1.15741E-05,
+          "PERMEABILITY_YY": 1.15741E-05,
+          "PERMEABILITY_XY": 0,
+          "DYNAMIC_VISCOSITY": 10000,
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/MaterialParameters_abc.json]: Now that we probably are in m, kg, sec and N should this be 0.0013? ( In DSettlement GUI permeability itself is given, which is translated to the input of both DYNAMIC_VISCOSITY and the intrinsic permeability PERMEABILITY_XX and _YY. So bringing it to 0.0013 maybe should imply also another PERMEABILITY_XX and _YY

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/MaterialParameters_linear.json [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+          "BIOT_COEFFICIENT": 1,
+          "DENSITY_SOLID": 2242.609582059123,
+          "DENSITY_WATER": 1019.367991845056,
+          "POROSITY": 0.5,
+          "BULK_MODULUS_SOLID": 20000000000,
+          "BULK_MODULUS_FLUID": 2200000000,
+          "PERMEABILITY_XX": 1.15741E-05,
+          "PERMEABILITY_YY": 1.15741E-05,
+          "PERMEABILITY_XY": 0,
+          "DYNAMIC_VISCOSITY": 10000,
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/MaterialParameters_linear.json]: Now that we probably are in m, kg, sec and N should this be 0.0013?

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/MaterialParameters_abc.json (hunk 18 above)
Comment (mnabideltares) [reply applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/MaterialParameters_abc.json]: Right, 10000 does not make much sense, However, their output from dsettlement and the provided analytical results seems to be base on these material parameters. Changing the viscosity alone to 0.0013 does change the results. What do you think? Do we ask them to provide us with the correct values?

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json (hunk 16 above)
Comment (mnabideltares) [reply applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json]: from dsettlment, they provided such names. Do we change the names in our side, unreleated to their naming system? I think we need to be consistent with their output and ask them to change their naming output?

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/MaterialParameters_abc.json (hunk 18 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/MaterialParameters_abc.json]: What seems to have happened is that the intrinsic permeability ( PERMEABILITY_XX and _YY ) have changed in with the same factor that takes DYNAMIC_VISCOSITY from 0.0013 to 10000. When change back DYNAMIC_VISCOSITY, PERMEABILITY_XX and PERMEABILITY_YY results should remain the same.

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json (hunk 16 above)
Comment (WPK4FEM) [reply applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json]: There is an ongoing action to get more comprehensible model part names written from DSettlement to the json and mdpa files. Walter is working on it. For tests in our testsuite, I would like them to be as small and clean as possible, showing only the model parts and processes that influence the results, nothing else.

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/Schematic.svg
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/Schematic.svg:1]: There is a symbol ( something like capital Delta, but then upside down and with underlines of different size)  for a phreatic line. For this picture, it may be important to understand that at the Phreaticline the pressure is fixed ( a Dirichlet condition for Pw)

File: applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json (hunk 16 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/ProjectParameters_stage1.json]: Thanks for clarifying the names!

**Review by rfaasse — APPROVED** [review]
Thanks for processing the review comments, for me this looks like it's ready to go!

**Review by WPK4FEM — APPROVED** [review]
